holoviews/core/data/image.py

```diff
@@ -0,0 +1,250 @@
+                data = data[::-1, :]
+        if not isinstance(data, np.ndarray) or data.ndim not in [2, 3]:
+            raise ValueError('ImageInterface expects a 2D array.')
+
+        return data, {'kdims':kdims, 'vdims':vdims}, kwargs
+
+
+    @classmethod
+    def shape(cls, dataset):
+        return cls.length(dataset), len(dataset.dimensions()),
+
+
+    @classmethod
+    def length(cls, dataset):
+        return np.product(dataset.data.shape)
+
+
+    @classmethod
+    def validate(cls, dataset):
+        pass
+
+    @classmethod
+    def redim(cls, dataset, dimensions):
+        return dataset.data
+
+    @classmethod
+    def reindex(cls, columns, kdims=None, vdims=None):
+        data = columns.data
+        if vdims is not None and vdims != columns.vdims and len(columns.vdims) > 1:
+            inds = [columns.get_dimension_index(vd)-columns.ndims for vd in vdims]
+            return data[:, :, inds] if len(inds) > 1 else data[:, :, inds[0]]
+        return data
+
+    @classmethod
+    def range(cls, obj, dim):
+        dim_idx = obj.get_dimension_index(dim)
+        if dim_idx in [0, 1] and obj.bounds:
+            l, b, r, t = obj.bounds.lbrt()
+            if dim_idx:
+                halfd = (1./obj.ydensity)/2.
+                drange = (b+halfd, t-halfd)
+            else:
+                halfd = (1./obj.xdensity)/2.
+                drange = (l+halfd, r-halfd)
+        elif 1 < dim_idx < len(obj.vdims) + 2:
+            dim_idx -= 2
+            data = np.atleast_3d(obj.data)[:, :, dim_idx]
+            drange = (np.nanmin(data), np.nanmax(data))
+        else:
+            drange = (None, None)
+        return drange
+
+    
+    @classmethod
+    def values(cls, dataset, dim, expanded=True, flat=True):
+        """
+        The set of samples available along a particular dimension.
+        """
+        dim_idx = dataset.get_dimension_index(dim)
+        if dim_idx in [0, 1]:
+            l, b, r, t = dataset.bounds.lbrt()
+            dim2, dim1 = dataset.data.shape[:2]
+            d1_half_unit = float(r - l)/dim1/2.
+            d2_half_unit = float(t - b)/dim2/2.
+            d1lin = np.linspace(l+d1_half_unit, r-d1_half_unit, dim1)
+            d2lin = np.linspace(b+d2_half_unit, t-d2_half_unit, dim2)
+            if expanded:
+                values = np.meshgrid(d2lin, d1lin)[abs(dim_idx-1)]
+                return values.flatten() if flat else values
+            else:
+                return d2lin if dim_idx else d1lin
+        elif dataset.ndims <= dim_idx < len(dataset.dimensions()):
+            # Raster arrays are stored with different orientation
+            # than expanded column format, reorient before expanding
+            if dataset.data.ndim > 2:
+                data = dataset.data[:, :, dim_idx-dataset.ndims]
+            else:
+                data = dataset.data
+            data = np.flipud(data)
+            return data.T.flatten() if flat else data
+        else:
+            return None
```

Comment:
Probably should be (or is already) handled at the Dataset level, I think it's for cdims.